stone/backends/python_type_stubs.py

```diff
@@ -334,7 +334,7 @@ def _generate_struct_class_init(self, ns, struct):
             field_name_reserved_check = fmt_var(field.name, True)
             field_type = self.map_stone_type_to_pep484_type(ns, field.data_type)
 
-            if not is_nullable_type(field.data_type):
```

Comment:
Why was this `not is_nullable_type` in the first place? Was that just a mistake?

Reply:
Yeah this is the line that has a lot of history behind. The purpose of this PR is basically roll back the change in this line. T329658 keeps track of it. The long story short is that #126 and #134  introduced changes that basically makes every constructor arg and getter return type optional. This is not ideal, and #131 removed the issue with the getters, but did not touch the issue with the constructor. This PR changes that by only allowing values with default and nullable values to be `Optional`.